src/aiida/engine/processes/workchains/restart.py

```diff
@@ -375,9 +417,20 @@ def inspect_process(self) -> Optional['ExitCode']:
             elif node.is_failed and last_report.exit_code.status != 0:
                 template = '{}<{}> failed but a handler detected an unrecoverable problem, aborting'
 
-            self.report(template.format(*report_args))
+            self.report(template.format(self.ctx.process_name, node.pk))
+
+            if last_report.exit_code.status != 0:
+                return last_report.exit_code
+
+            if pause_process:
+                self.report(
+                    f'Resetting the iteration counter(s) and pausing for inspection. You can resume execution using '
+                    f'`verdi process play {self.node.pk}`, or kill the work chain using '
+                    f'`verdi process kill {self.node.pk}`.'
+                )
+                self.pause(f"Paused for user inspection, see: 'verdi process report {self.node.pk}'")
 
-            return last_report.exit_code
```

Comment:
Here, the `BaseRestartWorkChain` used the fact that returning `ExitCode(0)` in a step of the outline doesn't really do anything as far as I can tell: it simply starts executing the next step. Might be there is not even any difference between returning that and `None`.

I think it's better to be explicit and return the exit code in case its exit status is nonzero, and return `None` otherwise.